### PR TITLE
[REVIEW] Add min and max bounds for random dataframe generator numeric types

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR provides the ability to pass in `min` and `max` bounds incase of numeric dtypes to the randon dataframe generator API.

### CODE REVIEW

File: python/cudf/cudf/testing/dataset_generator.py [hunk 1]
```diff
@@ -388,7 +388,12 @@ def rand_dataframe(
                     ColumnParameters(
                         cardinality=cardinality,
                         null_frequency=null_frequency,
-                        generator=int_generator(dtype=dtype, size=cardinality),
+                        generator=int_generator(
+                            dtype=dtype,
+                            size=cardinality,
+                            min_bound=meta.get("min_bound", None),
```
Comment (isVoid) [inline python/cudf/cudf/testing/dataset_generator.py:394]: Would it be better to put the default values inline here instead of below?

**Review by isVoid — APPROVED** [review]
Verified to my branch that this works! Just one small comment - feel free to ignore at your discretion. Thanks for putting it together so quickly!

File: python/cudf/cudf/testing/dataset_generator.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/testing/dataset_generator.py:394]: We are going to use those `int_generator`, `float_generator` call's elsewhere in list generation and struct generation too. So inlining them here will force us to inline them elsewhere too.

**Comment by galipremsagar** [comment]
rerun tests

**Comment by isVoid** [comment]
rerun tests

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**